hphp/runtime/ext/pdo/ext_pdo.php

```diff
@@ -1,6 +1,15 @@
 <?hh
 // generated by idl-to-hni.php
 
+/**
+ * Objects implementing PDOEntity can handle value by column name and meta
+ */
+interface PDOEntity {
```

Comment:
I see that this feature is not yet in PHP either. Has there been any discussion on the internals list about it? I'm not opposed to us adding it without waiting for PHP support, though if they end up using a different interface we will need to change ours to match what they implement. Until this feature is in PHP we will probably want this interface to live in the `HH` namespace.

I'm not entirely sure I follow the rationale in your commit summary, could you add some more detail?

https://bugs.php.net/bug.php?id=69016